skrifa/src/outline/autohint/latin/edges.rs

```diff
@@ -244,27 +307,42 @@ pub(crate) fn compute_blue_edges(
             // Both directions are handled for neutral blues
             if is_top ^ is_major_dir || is_neutral {
                 // Compare to reference position
-                let dist = fixed_mul(
-                    (edge.fpos as i32 - unscaled_blue.position).abs(),
-                    scale.y_scale,
-                );
+                let (ref_pos, matching_blue) = if group == ScriptGroup::Default {
+                    (unscaled_blue.position, blue.position)
+                } else {
+                    // For CJK, we take the blue with the smallest delta
+                    // from the edge
+                    // See <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/autofit/afcjk.c#L1356>
+                    if (edge.fpos as i32 - unscaled_blue.position).abs()
+                        > (edge.fpos as i32 - unscaled_blue.overshoot).abs()
```

Comment:
I suggest adding some methods to things like `ScaledBlue` and `Edge`. For example, instead of `let is_top = blue.flags & (blue_flags::TOP | blue_flags::LATIN_SUB_TOP) != 0;` we could have `blue.is_top()`, instead of `(edge.fpos as i32 - unscaled_blue.position).abs()` we could have `edge.distance_to(unscaled_blue)` (...or would it need to be specifically unscaled_distance_to?)

This could be deferred to an issue if we don't want to address here. I feel like autohinting would be quite a bit easier to read if widely deployed.

Reply:
There's a lot that can be done to clean up the whole autohint module but I prioritized (mostly) matching the FT code in the first pass. I agree that we can do much better but I would like to defer until after we land. Filed #1141